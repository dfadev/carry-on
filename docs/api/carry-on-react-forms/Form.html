<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>Form · carry-on</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="State container for React"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="Form · carry-on"/><meta property="og:type" content="website"/><meta property="og:url" content="https://dfadev.github.io//carry-on/index.html"/><meta property="og:description" content="State container for React"/><meta property="og:image" content="https://dfadev.github.io//carry-on/img/docusaurus.png"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://dfadev.github.io//carry-on/img/docusaurus.png"/><link rel="shortcut icon" href="/carry-on/img/favicon.ico"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"/><link rel="stylesheet" href="/carry-on/css/main.css"/></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/carry-on/"><img class="logo" src="/carry-on/img/card-travel.svg" alt="carry-on"/><h2 class="headerTitleWithLogo">carry-on</h2></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class="siteNavGroupActive"><a href="/carry-on/docs/" target="_self">Overview</a></li><li class="siteNavGroupActive"><a href="/carry-on/docs/api/" target="_self">API</a></li><li class=""><a href="https://github.com/dfadev/carry-on" target="_self">GitHub</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="container docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><i></i></div><h2><i>›</i><span>API</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">Overview</h3><ul><li class="navListItem"><a class="navItem" href="/carry-on/docs/">Getting Started</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/registeringState">Registering State</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/usingState">Using State</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/examples/">Examples</a></li></ul></div><div class="navGroup"><h3 class="navGroupCategoryTitle">API</h3><ul><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/">Info</a></li><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">carry-on-store</h4><ul><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-store/deleteStore">deleteStore</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-store/devTools">devTools</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-store/initStores">initStores</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-store/register">register</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-store/subscribe">subscribe</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-store/useStore">useStore</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">carry-on-react</h4><ul><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-react/State">&lt;State ...&gt;</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-react/withState">withState</a></li></ul></div><div class="navGroup subNavGroup"><h4 class="navGroupSubcategoryTitle">carry-on-react-forms</h4><ul><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-react-forms/Field">Field</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-react-forms/form">form</a></li><li class="navListItem navListItemActive"><a class="navItem" href="/carry-on/docs/api/carry-on-react-forms/Form">Form</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-react-forms/FormButtons">FormButtons</a></li><li class="navListItem"><a class="navItem" href="/carry-on/docs/api/carry-on-react-forms/FormState">FormState</a></li></ul></div></ul></div></div></section></div><script>
            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              const headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                if (event.target.tagName === 'A') {
                  document.body.classList.remove('tocActive');
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer"><div class="wrapper"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle">Form</h1></header><article><div><span></span></div></article></div><div class="docs-prevnext"><a class="docs-prev button" href="/carry-on/docs/api/carry-on-react-forms/form"><span class="arrow-prev">← </span><span>form</span></a><a class="docs-next button" href="/carry-on/docs/api/carry-on-react-forms/FormButtons"><span class="function-name-prevnext">FormButtons</span><span class="arrow-next"> →</span></a></div></div></div><nav class="onPageNav"></nav></div><footer class="nav-footer" id="footer"><section class="copyright">Copyright © 2019 Russ Panula</section></footer></div></body></html>