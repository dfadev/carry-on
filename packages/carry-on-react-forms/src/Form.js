/** @format **/
import React from "react";
import { State } from "carry-on-react";

export default ({ store, form = "form", children, ...rest }) => (
  <State path={form} from={store} {...rest}>
    {state => (
      <form onSubmit={state.submit} onReset={state.reset}>
        {children}
      </form>
    )}
  </State>
);
