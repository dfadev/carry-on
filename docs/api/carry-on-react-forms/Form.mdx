---
id: Form
title: Form
---
import BrowserOnly from "@docusaurus/BrowserOnly";
import Inspector from "react-inspector";
import { FormState } from "carry-on-react-forms";
import { State } from "carry-on-react";
import theme from "../../inspector-theme";

## Import

```js

import { Form } from "carry-on-react-forms";

```

## `Form`


```js live noInline
State.Debug = true;
State.Verbose = true;

const formStateFactory = (formId = "form") => ({
  state: ({ get, set }) => ({
    [formId]: {
      initialValues: {
        field1: "value1",
        field2: "value2"
      },
      onSubmit(vals) {
        alert(JSON.stringify(vals));
      }
    }
  })
});

render(
  <Form id="myForm" register={formStateFactory("myForm")}>
    <div>
      <Field path="field1">
        {field => <input {...field.element} />}
      </Field>
    </div>
    <div>
      <Field path="field2">
        {field => <input {...field.element} />}
      </Field>
    </div>
    <div>
      <button type="submit">OK</button>
    </div>
  </Form>
);
```

<BrowserOnly>
  {() => {
    <State select={s => ({ ...s })}>
      {state => <Inspector data={state} theme={theme} expandLevel={2} />}
    </State>
  }}
</BrowserOnly>
