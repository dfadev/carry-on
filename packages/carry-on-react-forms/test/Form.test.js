import React from "react";
import { render } from "react-testing-library";
import Form from "../src/Form";

it("should render <Form>", () => {
  expect(render(<Form />).asFragment()).toMatchSnapshot();
});
