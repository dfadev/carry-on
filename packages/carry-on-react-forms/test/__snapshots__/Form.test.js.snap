// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles initialValues specified as a function 1`] = `
Object {
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "isPristine": true,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": false,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "some": "stuff",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "some": "stuff",
    },
    "visited": Object {},
  },
}
`;

exports[`handles initialValues specified as prop 1`] = `
Object {
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "isPristine": true,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": false,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "some": "stuff",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "some": "stuff",
    },
    "visited": Object {},
  },
}
`;

exports[`renders store id 1`] = `
<DocumentFragment>
  <form
    id="zzz.123"
  />
</DocumentFragment>
`;

exports[`should handle reset with no onReset 1`] = `
Object {
  "checkbox": false,
  "field1": "value1",
  "field2": "value2",
}
`;

exports[`should handle setValues using same values 1`] = `
Object {
  "checkbox": false,
  "field1": "value1",
  "field2": "value2",
}
`;

exports[`should render <Form> 1`] = `
<DocumentFragment>
  <form
    id="form"
  />
</DocumentFragment>
`;

exports[`should validate: prop match 1`] = `
Object {
  "element": Object {
    "id": "default.form.field1",
    "name": "default.form.field1",
    "onBlur": [Function],
    "onChange": [Function],
    "onFocus": [Function],
    "type": "text",
    "value": "value1",
  },
  "error": undefined,
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "initialValues": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "isPristine": true,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": false,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "checkbox": false,
        "field1": "value1",
        "field2": "value2",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "visited": Object {},
  },
  "setError": [Function],
  "setTouched": [Function],
  "setValue": [Function],
  "setVisited": [Function],
  "touched": false,
  "visited": false,
}
`;

exports[`should validate: prop match 2`] = `
Object {
  "element": Object {
    "id": "default.form.field1",
    "name": "default.form.field1",
    "onBlur": [Function],
    "onChange": [Function],
    "onFocus": [Function],
    "type": "text",
    "value": "123",
  },
  "error": undefined,
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "initialValues": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "isPristine": false,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": true,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "checkbox": false,
        "field1": "value1",
        "field2": "value2",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "checkbox": false,
      "field1": "123",
      "field2": "value2",
    },
    "visited": Object {},
  },
  "setError": [Function],
  "setTouched": [Function],
  "setValue": [Function],
  "setVisited": [Function],
  "touched": false,
  "visited": false,
}
`;

exports[`should validate: prop match 3`] = `
Object {
  "element": Object {
    "id": "default.form.field1",
    "name": "default.form.field1",
    "onBlur": [Function],
    "onChange": [Function],
    "onFocus": [Function],
    "type": "text",
    "value": "123",
  },
  "error": undefined,
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "initialValues": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "isPristine": false,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": false,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "checkbox": false,
        "field1": "value1",
        "field2": "value2",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "checkbox": false,
      "field1": "123",
      "field2": "value2",
    },
    "visited": Object {},
  },
  "setError": [Function],
  "setTouched": [Function],
  "setValue": [Function],
  "setVisited": [Function],
  "touched": false,
  "visited": false,
}
`;

exports[`should validate: prop match 4`] = `
Object {
  "element": Object {
    "id": "default.form.field1",
    "name": "default.form.field1",
    "onBlur": [Function],
    "onChange": [Function],
    "onFocus": [Function],
    "type": "text",
    "value": "345",
  },
  "error": undefined,
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "initialValues": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "isPristine": false,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": true,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "checkbox": false,
        "field1": "value1",
        "field2": "value2",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "checkbox": false,
      "field1": "345",
      "field2": "value2",
    },
    "visited": Object {},
  },
  "setError": [Function],
  "setTouched": [Function],
  "setValue": [Function],
  "setVisited": [Function],
  "touched": false,
  "visited": false,
}
`;

exports[`should validate: prop match 5`] = `
Object {
  "element": Object {
    "id": "default.form.field1",
    "name": "default.form.field1",
    "onBlur": [Function],
    "onChange": [Function],
    "onFocus": [Function],
    "type": "text",
    "value": "345",
  },
  "error": undefined,
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "initialValues": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "isPristine": false,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": false,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "checkbox": false,
        "field1": "value1",
        "field2": "value2",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "checkbox": false,
      "field1": "345",
      "field2": "value2",
    },
    "visited": Object {},
  },
  "setError": [Function],
  "setTouched": [Function],
  "setValue": [Function],
  "setVisited": [Function],
  "touched": false,
  "visited": false,
}
`;

exports[`should validate: render match 1`] = `
<DocumentFragment>
  <form
    id="form"
  >
    <input
      id="default.form.field1"
      name="default.form.field1"
      type="text"
      value="value1"
    />
  </form>
</DocumentFragment>
`;

exports[`should validate: store match 1`] = `
Object {
  "form": Object {
    "errors": Object {},
    "formId": "form",
    "hasError": [Function],
    "hasVisited": [Function],
    "initialValues": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "isPristine": true,
    "isSubmitting": false,
    "isTouched": [Function],
    "isValid": true,
    "isValidating": false,
    "origState": Object {
      "errors": Object {},
      "formId": "form",
      "hasError": [Function],
      "hasVisited": [Function],
      "isPristine": true,
      "isSubmitting": false,
      "isTouched": [Function],
      "isValid": true,
      "isValidating": false,
      "reset": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFieldVisited": [Function],
      "setInitialValues": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submit": [Function],
      "submitCount": 0,
      "touched": Object {},
      "validate": [Function],
      "values": Object {
        "checkbox": false,
        "field1": "value1",
        "field2": "value2",
      },
      "visited": Object {},
    },
    "reset": [Function],
    "setErrors": [Function],
    "setFieldError": [Function],
    "setFieldTouched": [Function],
    "setFieldValue": [Function],
    "setFieldVisited": [Function],
    "setInitialValues": [Function],
    "setTouched": [Function],
    "setValues": [Function],
    "submit": [Function],
    "submitCount": 0,
    "touched": Object {},
    "validate": [Function],
    "values": Object {
      "checkbox": false,
      "field1": "value1",
      "field2": "value2",
    },
    "visited": Object {},
  },
}
`;
